src/enzyme_ad/jax/Passes/CanonicalizeFor.cpp

```diff
@@ -908,6 +908,120 @@ struct WhileToForHelper {
   }
 };
 
+//This works fine for doWhile with no iter_args case only as of now
+//A more proper approach: copy before region to outside for loop, and copy both before and after region to inside for loop ?
+struct MoveDoWhileToFor : public OpRewritePattern<WhileOp> {
+  using OpRewritePattern<WhileOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(WhileOp whileOp,
+                                PatternRewriter &rewriter) const override {
+      auto module = whileOp->getParentOfType<ModuleOp>();
+      // 1. Trace scf.condition to get condition -> condition is arith.cmpi -> From condition get UB.
```

Comment:
can you check explicitly here tht there are no iter_args and return failure if so (rather than risk incorrectness/crash)